### ISSUE TITLE
[FEATURE] Xposed-only version

### ISSUE THREAD

**Issue by k2helix** [Mahmud0808/Iconify#166]
### Describe the feature

It would be nice to have a xposed-only version of Iconify which exclusively includes the xposed tweaks without the need of the functions that require the magisk module to be installed.

### Why would it be useful to add?

It would be useful as people who only use the xposed tweaks for things such as battery icons, transparency and blur, clocks, etc. can install a "lite" version of iconify which doesn't need a magisk module which provides all the other features. Some other customisation apps such as AOSPMods already have this feature.

### Example(s)

Can't really give examples, but a use case would be the one stated above: using iconify just for xposed tweaks such as battery icon or more

**Comment by Mahmud0808** [comment]
A few of the xposed features need the magisk module for showing correct colors on background chips. That's why I kept module installation mandatory.

**Comment by k2helix** [comment]
I see. Maybe let the user know what can they do with only Xposed first? Or almost all the features need the magisk module?

**Comment by Mahmud0808** [comment]
> Or almost all the features need the magisk module?

No only one or two.

Okay I will need to figure out a convenient method to skip module installation and directly redirect to xposed page. Thanks for the suggestion.

**Comment by k2helix** [comment]
Thanks for your reply! I guess that you can either do it in the same app and let the user choose or just provide another apk with it.

I don't know whether I should close this issue, so feel free to do it if that's the case.

**Comment by k2helix** [comment]
Hi again and thank you for implementing this feature. 

I've seen in the changelog that to access the xposed only version I need to hold the Let's go button. However this crashes the app.

It has the needed xposed scopes (system ui and system framework). I did not give it root access as if it is xposed only there should be no need of it. Do I need to grant it?

Thanks!

**Comment by Mahmud0808** [comment]
Yes you need root, and also share crash log.

**Comment by k2helix** [comment]
```
FATAL EXCEPTION: AsyncTask #1
Process: com.drdisagree.iconify, PID: 20786
java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$4.done(AsyncTask.java:415)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:305)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:920)
Caused by: org.zeroturnaround.zip.ZipException: Given file '/storage/emulated/0/.iconify/Iconify' doesn't exist!
	at org.zeroturnaround.zip.ZipUtil.pack(ZipUtil.java:1616)
	at org.zeroturnaround.zip.ZipUtil.pack(ZipUtil.java:1413)
	at org.zeroturnaround.zip.ZipUtil.pack(ZipUtil.java:1396)
	at com.drdisagree.iconify.ui.activities.OnBoardingScreen3$startInstallationProcess.doInBackground(OnBoardingScreen3.java:420)
	at com.drdisagree.iconify.ui.activities.OnBoardingScreen3$startInstallationProcess.doInBackground(OnBoardingScreen3.java:212)
	at android.os.AsyncTask$3.call(AsyncTask.java:394)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 4 more
```

Then the difference between the xposed only version is that it does not need the magisk module, but requires root too

**Comment by Mahmud0808** [comment]
> Then the difference between the xposed only version is that it does not need the magisk module, but requires root too

A few features like QS margins, restarting systemui need root access. Also in xposed-only mode it installs an empty module as it's also needed for writing some shell scripts (for QS margins) which will automatically execute after phone boots up.

The main difference is you won't have hundreds of overlays installed.

Edit: I suppose you wanted something which could work without root. Like LSPatch or something?

**Comment by k2helix** [comment]
> Edit: I suppose you wanted something which could work without root. Like LSPatch or something?

No I just meant using LSPosed only without any kind of magisk module or granting the app root access. I don't mean something that could work without the phone being rooted, probably that's impossible. Just installing the app, giving it the scopes it needs in LSPosed, and being able to use the Xposed features of Iconify.

This is a nice implementation as now it can be installed without all the overlays, but I originally meant that.

**Comment by Mahmud0808** [comment]
> No I just meant using LSPosed only without any kind of magisk module or granting the app root access.

Forgot to mention, the blur enabler also needs magisk module. So I guess I will not do a pure xposed only version as some important features are gonna get cutoff.

> This is a nice implementation as now it can be installed without all the overlays, but I originally meant that.

Can you try the latest debug and let me know if it works now? I can't reproduce the issue you are having.

**Comment by k2helix** [comment]
Well it's fine, don't worry about it if a lot of features cannot work with a pure xposed version.

I tried the Debug APK, gave it lsposed access to System UI and System Framework, but as I did with the stable version I didn't grant it root access.

It now does not crash and warns about some error saved in the logs folder which says that it cannot create the module file.
![Screenshot_20230708-210340_Iconify (Debug).png](https://github.com/Mahmud0808/Iconify/assets/49949593/1f1c77f6-bac8-4d05-b700-0b343e266ef5)

But as I said this is by not granting it superuser access. By doing it I don't think this error happens at all.
